apps/api/src/domain/models/song/queries/song.view-model.event.sourced.ts

```diff
@@ -0,0 +1,115 @@
+import { FromDomainModel, NonNegativeFiniteNumber, URL } from '@coscrad/data-types';
+import { isNonEmptyObject, isNonEmptyString } from '@coscrad/validation-constraints';
+import { CoscradDataExample } from '../../../../test-data/utilities';
+import { DTO } from '../../../../types/DTO';
+import { buildMultilingualTextWithSingleItem } from '../../../common/build-multilingual-text-with-single-item';
+import { MultilingualText } from '../../../common/entities/multilingual-text';
+import buildDummyUuid from '../../__tests__/utilities/buildDummyUuid';
+import { AccessControlList } from '../../shared/access-control/access-control-list.entity';
+import { ContributionSummary } from '../../user-management';
+import { SongCreated } from '../commands';
+import { Song } from '../song.entity';
+
+const FromSong = FromDomainModel(Song);
+
+const testEventId = buildDummyUuid(1);
+
+@CoscradDataExample<EventSourcedSongViewModel>({
+    example: {
+        id: testEventId,
+        name: buildMultilingualTextWithSingleItem('test song'),
+        isPublished: false,
+        accessControlList: new AccessControlList(),
+        contributions: [],
+        lyrics: buildMultilingualTextWithSingleItem('lalala'),
+        audioURL: 'https://www.coscrad.org/lalala.mp3',
+        lengthMilliseconds: 1500,
+    },
+})
+export class EventSourcedSongViewModel {
+    // TODO share these props with other resource views
```

Comment:
I'll do a follow up PR after yours to start sharing code. We've got enough of these to find the right pattern now.